Example/PrebidDemoJava/src/main/java/org/prebid/mobile/app/RubiconInstreamVideoDemoActivity.java

```diff
@@ -0,0 +1,164 @@
+
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.google.android.exoplayer2.SimpleExoPlayer;
+import com.google.android.exoplayer2.ext.ima.ImaAdsLoader;
+import com.google.android.exoplayer2.source.MediaSource;
+import com.google.android.exoplayer2.source.ProgressiveMediaSource;
+import com.google.android.exoplayer2.source.ads.AdsMediaSource;
+import com.google.android.exoplayer2.ui.PlayerView;
+import com.google.android.exoplayer2.upstream.DataSource;
+import com.google.android.exoplayer2.upstream.DefaultDataSourceFactory;
+
+import org.prebid.mobile.AdUnit;
+import org.prebid.mobile.Host;
+import org.prebid.mobile.OnCompleteListener2;
+import org.prebid.mobile.PrebidMobile;
+import org.prebid.mobile.ResultCode;
+import org.prebid.mobile.Signals;
+import org.prebid.mobile.Util;
+import org.prebid.mobile.VideoAdUnit;
+import org.prebid.mobile.VideoBaseAdUnit;
+
+import java.util.Arrays;
+import java.util.Map;
+
+public class RubiconInstreamVideoDemoActivity extends AppCompatActivity {
+    AdUnit adUnit;
+    private PlayerView playerView;
+    private SimpleExoPlayer player;
+    private ImaAdsLoader adsLoader;
+
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+        if (adUnit != null) {
+            adUnit.stopAutoRefresh();
+            adUnit = null;
+        }
+        adsLoader.release();
+    }
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_instream_video);
+        playerView = findViewById(R.id.player_view);
+
+        PrebidMobile.setPrebidServerHost(Host.RUBICON);
+        PrebidMobile.setPrebidServerAccountId(Constants.PBS_ACCOUNT_ID_RUBICON);
+        PrebidMobile.setStoredAuctionResponse(Constants.PBS_STORED_RESPONSE_VAST_RUBICON);
+        VideoBaseAdUnit.Parameters parameters = new VideoBaseAdUnit.Parameters();
+        parameters.setMimes(Arrays.asList("video/mp4"));
+
+        parameters.setProtocols(Arrays.asList(Signals.Protocols.VAST_2_0));
+        // parameters.setProtocols(Arrays.asList(new Signals.Protocols(2)));
+
+        parameters.setPlaybackMethod(Arrays.asList(Signals.PlaybackMethod.AutoPlaySoundOff));
+        // parameters.setPlaybackMethod(Arrays.asList(new Signals.PlaybackMethod(2)));
+
+        parameters.setPlacement(Signals.Placement.InStream);
+        // parameters.setPlacement(new Signals.Placement(2));
+
+        VideoAdUnit adUnit = new VideoAdUnit("1001-1", 300, 250);
+        adUnit.setParameters(parameters);
+        this.adUnit = adUnit;
+        // Create an AdsLoader with the ad tag url.
+        adUnit.fetchDemand(new OnCompleteListener2() {
+            @Override
+            public void onComplete(ResultCode resultCode, Map<String, String> unmodifiableMap) {
+                String uri = Util.generateInstreamUriForGam("/19968336/Wei_instream_video", 640, 480, unmodifiableMap);
```

Comment:
I have to create a Rubicon adUnitId and replace it

Reply:
You can update the PR directly.